src/tz.cpp

```diff
@@ -2156,27 +2163,29 @@ time_zone::get_info_impl(sys_seconds tp) const
                                      [](const sys_seconds& x, const transition& t)
                                      {
                                          return x < t.timepoint;
-                                     }));
+                                     }), tp);
 }
 
 local_info
 time_zone::get_info_impl(local_seconds tp) const
 {
     using namespace std::chrono;
     init();
-    local_info i;
+    local_info i; 
     i.result = local_info::unique;
     auto tr = upper_bound(transitions_.begin(), transitions_.end(), tp,
                           [](const local_seconds& x, const transition& t)
                           {
                               return sys_seconds{x.time_since_epoch()} -
                                                          t.info->offset < t.timepoint;
                           });
-    i.first = load_sys_info(tr);
+    if (tr == transitions_.cend()) return Posix::time_zone{ posix_timezone_ }.get_info(tp);//TODO This logic needs updating for some single transition TZ's
+    auto sys_tp = sys_seconds{ tp.time_since_epoch() } - tr->info->offset;
+    i.first = load_sys_info(tr, sys_tp);
```

Comment:
This logic needs extensive testing, but is correct enough to pass mild scrutiny. We get a chicken and egg problem where to get the UTC timepoint we need to calculate the UTC timepoint.

This code isn't needed for either workflow *yet*, so I kept as simple as would compile.